server/app/models/suggestquery.rb

```diff
@@ -0,0 +1,66 @@
+class Suggestquery
+  def initialize(query_key)
+    @query_key = query_key
+  end
+
+  def find
+    find_record_by_query_key
+  end
+
+  private
+
+  attr_reader :query_key
+
+  def find_record_by_query_key
+    result = nil
+    result = find_block_by_number if integer_string?
+    result = find_by_hex if valid_hex?
```

Comment:
A little confusing here

may `result = if .... else ...` better here

Reply:
Tank you. I will change it.